src/ORM/Rule/ExistsIn.php

```diff
@@ -98,6 +109,11 @@ public function __invoke(EntityInterface $entity, array $options)
             return true;
         }
 
+        if ($this->_options['allowSqlNulls'] === true
+            && $this->_checkPartialSchemaNulls($entity, $source) === true
+        ) {
+            return true;
```

Comment:
@ionas Indeed you did it in the first commit. I simply think that if the current behavior is a bug, we can fix it in 3.x. If it's a feature then 4.x might be suitable as you said.

At the moment, I cannot understand what they are telling me at all, if they are speaking English, not writing :stuck_out_tongue: Furthermore I have never been abroad. But I am planning to join next CakeFest. I hope to see you next time :cake: